File: 13/13.cpp
#include <iostream>
using namespace std;
int findSecondLargest(int arr[], int size) {
    int largest = arr[0];
    int secondLargest = arr[0];
    for (int i = 1; i < size; i++) {
        if (arr[i] > largest) {
            secondLargest = largest;
            largest = arr[i];
        } else if (arr[i] > secondLargest && arr[i] != largest) {
            secondLargest = arr[i];
        }
    }
    return secondLargest;
}

int main() {
    int size;
    cout<<"Enter the size of the array: ";
    cin>>size;
    
    int arr[size];
    cout<<"Enter "<<size<<" elements of the array:"<<endl;
    for (int i = 0; i < size; i++) {
        cin>>arr[i];
    }
    
    int secondLargest = findSecondLargest(arr, size);
    cout<<"The second largest element in the array is:"<<secondLargest;
    return 0;
}

File: 13/13.py
def second_largest(arr):
    largest = second_largest = float('-inf')
    for num in arr:
        if num > largest:
            second_largest = largest
            largest = num
        elif num > second_largest and num != largest:
            second_largest = num
    if second_largest == float('-inf'):
        return "There is no second largest element"
    else:
        return second_largest

n = int(input("Enter the number of elements in the array: "))
arr = []
for i in range(n):
    element = int(input(f"Enter element {i+1}: "))
    arr.append(element)
result = second_largest(arr)
print("The second largest element in the array is:", result)

